partner_preview_scripts/encore_parser.py

```diff
@@ -0,0 +1,610 @@
+        lfc_df = self.get_lfc_data(lfc_file)
+
+        logging.info(
+            f'Number of gene pairs in the log(fold change) dataset: {lfc_df.select("id").distinct().count()}'
+        )
+        logging.info(
+            f'Number cell lines in the log(fold change) dataset: {lfc_df.select("cellLineName").distinct().count()}'
+        )
+        # A decistion was made to exclude gemini asessments.
+        # # Reading gemini data:
+        # gemini_file = f"{self.data_folder}/{gemini_file}"
+        # gemini_df = self.get_gemini_data(gemini_file)
+        # logging.info(
+        #     f'Number of gene pairs in the gemini dataset: {gemini_df.select("id").distinct().count()}'
+        # )
+        # logging.info(
+        #     f'Number cell lines in the gemini dataset: {gemini_df.select("cellLineName").distinct().count()}'
+        # )
+
+        # """WARNING: Based on the communication with the encore team, the BLISS dataset is currently considered unreliable."""
+        bliss_file = f"{self.data_folder}/{blissFile}"
+        bliss_df = self.get_bliss_data(bliss_file)
+
+        # Merging lfc + gemini:
+        merged_dataset = (
+            lfc_df
+            # Data is joined by the gene-pair and cell line:
+            # .join(bliss_df, how='inner', on=['id', 'cellLineName'])
+            .join(bliss_df, how="inner", on=["id", "cellLineName"])
+            # Applying filters on logFoldChange + interaction p-value thresholds:
+            .filter(
+                (
+                    f.col("phenotypicConsequencePValue")
+                    <= self.log_fold_change_cutoff_p_val
+                )
+                & (f.col("phenotypicConsequenceFDR") <= self.log_fold_change_cutoff_FDR)
+                & (f.col("geneticInteractionPValue") <= self.interaction_cutoff_p_val)
+            )
+            # Cleaning the cell line annotation:
+            .withColumn("cellId", f.split(f.col("cellLineName"), "_").getItem(0))
+            # Joining with cell passport data containing diseaseCellLine and biomarkers info:
+            .join(
+                self.cell_passport_df.select(
+                    f.col("id").alias("cellId"), "diseaseCellLine", "biomarkerList"
+                ),
+                on="cellId",
+                how="left",
+            ).persist()
+        )
+        logging.info(
+            f'Number of gene pairs in the merged dataset: {merged_dataset.select("id").count()}'
+        )
+        logging.info(
+            f'Number of cell lines in the merged dataset: {merged_dataset.select("cellLineName").distinct().count()}'
+        )
+
+        evidence_df = (
+            merged_dataset.withColumn(
+                "diseaseCellLines", f.array(f.col("diseaseCellLine"))
+            )
+            # Parsing/exploding gene names and target roles:
+            .withColumn("id", self.parse_targets(f.col("id"), f.col("Note1")))
+            .select("*", f.explode(f.col("id")).alias("genes"))
```

Comment:
```suggestion
            .select("*", f.explode(self.parse_targets(f.col("id"), f.col("Note1"))).alias("genes"))
```

Reply:
Based on the discussion I have with Emre last year they will change the format of the files, so optimizing this implementation doesn't make much sense.